### ISSUE TITLE
Clean-up code

### ISSUE THREAD

**Issue by rb643** [autism-research-centre/Autism-Gradients#8]
Now its all in one long script, section need to be cleanly parcellated to different notebooks so its easier to update and debug at different stages by different people. 

Also save the output intermediately so that the next notebook can actually start from there.

**Comment by rb643** [comment]
My suggestions for filenames and breaking it up:
- 1_SelectFiles; select the correct files based on missing time points and phenotype file
- 2_DiffusionEmbedding; run diffusion embedding on selected files from 1
- 3_Backprojection; back-project the diffusion embeddings
- 4 Quality Control; 
- 4a_Plot_Mean; plot the mean vectors
- 4b_Plot_Indiv; plot some individuals and write them out as Nifti
- 4b_Plot_Indiv_bins; plot the binned individual gradients and write them out as Nifti
- 5 Gradient Slope
- 5a_ GradientSlopes; obtain the gradient slopes and write them to phenotype csv
- 5b_ VisualizeSlopes; to have a visual inspection of some of the fits
- 5c_ CompareGroups; to compare and plot the gradient slopes per group
- 6 Goodness of fit analysis
- 6a_GoF_Keywords; to compare the primary gradient against some known networks
- 6b_GoF_Binned; to compare the different bins within the primary gradient

Let me know if I'm forgetting any steps?

**Comment by rb643** [comment]
ps: 1, 2 and 3 are done

**Comment by rb643** [comment]
4 and 5 done

**Comment by janfreyberg** [comment]
I'm working on 6

**Comment by rb643** [comment]
Excellent! Cause I'm still having issues setting up the matlab kernel. Just pip installing the kernel doesn't seem enough... Do you somehow have to point it to your Matlab installation somewhere as well? I can't pick a matlab notebook in jupyter.

**Comment by rb643** [comment]
ps: when you're script is ready you can save it as a markdown filed store it in the _posts/notebooks/ folder using YYYY-MM-DD- as the prefix and add the following to the top:
---
layout:     post
title:      "Step 6 - Setup your environment"
subtitle:   "give some description"
date:       2017-03-05 12:00:00
author:     "Jan Freyberg"
header-img: "img/home-bg.jpg"
categories: notebooks
---

The site should then pick it up automatically

**Comment by janfreyberg** [comment]
Yes, that's a good point! You actually do need to do a bit more:
https://uk.mathworks.com/help/matlab/matlab_external/install-the-matlab-engine-for-python.html

**Comment by janfreyberg** [comment]
Forgot about this step

**Comment by rb643** [comment]
Any follow up steps? I still can't pick the matlab kernel.. :(

**Comment by janfreyberg** [comment]
What happens when you run
```
import matlab
matlab.start_engine()
```
in python?

**Comment by rb643** [comment]
>>> matlab.start_engine()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'start_engine'
>>>

**Comment by janfreyberg** [comment]
Sorry, wrong code:
```
import matlab.engine
eng = matlab.engine.start_matlab()
```

**Comment by rb643** [comment]
>>> import matlab.engine
>>> eng = matlab.engine.start_matlab()
>>>

**Comment by rb643** [comment]
Do I need to run that prior to running jupiter notebooks?

**Comment by janfreyberg** [comment]
No, just checking that the matlab engine is installed properly.

I think last step might be:

`python -m matlab_kernel install`

**Comment by janfreyberg** [comment]
(this should hopefully register the kernel with jupyter, so then when you open jupyter notebook you'll see the different kernels)

**Comment by rb643** [comment]
Excellent! Works now!
